Common/CUDA/tvdenoising.cu

```diff
@@ -369,11 +380,12 @@ do { \
                     is_last_chunk=!((sp*deviceCount+dev)<deviceCount*splits-1);
                     is_first_chunk=!(sp*deviceCount+dev);
                     
-                    // lets compute where we start copyes and how much. This avoids 3 calls to Memcpy
+                    // lets compute where we start copies and how much. This avoids 3 calls to Memcpy
                     offset_device[dev]=buffer_pixels*is_first_chunk;
                     offset_host[dev]=linear_idx_start-buffer_pixels*!is_first_chunk;
                     bytes_device[dev]=curr_pixels+buffer_pixels*!is_first_chunk+buffer_pixels*!is_last_chunk;
                 }
+                // copy data to the GPU if we are just starting
                 if(i==0){
                     for (dev = 0; dev < deviceCount; dev++){

```

Comment:
Hello @AnderBiguri . In the latest update of the code, the part of the code for when i==0 doesn't seem to have been updated! Otherwise, there will still be a ghost. the new code should be here:
```
for (dev = 0; dev < deviceCount; dev++){
                        is_last_chunk=!((sp*deviceCount+dev)<deviceCount*splits-1);
                        is_first_chunk=!(sp*deviceCount+dev);

                        cudaSetDevice(gpuids[dev]);
                        if (is_last_chunk) {cudaMemsetAsync(d_src[dev], 0, mem_img_each_GPU,stream[dev*nStream_device+1]);}
                        cpy_from_host(d_src[dev],src,bytes_device[dev], offset_device[dev],offset_host[dev], pixels_per_slice, buffer_length, stream[dev*nStream_device+1],  is_first_chunk,  is_last_chunk, image_size);
                    }
                    for (dev = 0; dev < deviceCount; dev++){
                        cudaSetDevice(gpuids[dev]);

                        cudaMemcpyAsync(d_u[dev], d_src[dev], mem_img_each_GPU, cudaMemcpyDeviceToDevice,stream[dev*nStream_device+1]);
                        cudaMemsetAsync(d_px[dev], 0, mem_img_each_GPU,stream[dev*nStream_device]);
                        cudaMemsetAsync(d_py[dev], 0, mem_img_each_GPU,stream[dev*nStream_device]);
                        cudaMemsetAsync(d_pz[dev], 0, mem_img_each_GPU,stream[dev*nStream_device]);
                    }
```

Reply:
Thanks @zezisme  and apologies, I am doing 17 things at the same time, so looking at this with limited brainpower.